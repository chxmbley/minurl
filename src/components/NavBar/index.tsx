import Link from 'next/link';
import type { FC } from 'react';

/** Application navigation bar */
const NavBar: FC = () => (
  <nav className="flex justify-between py-4">
    <Link href="/" passHref>
      <a>
        <h1 className="text-2xl font-bold text-slate-700 dark:text-slate-200">minurl ðŸ’Ž</h1>
      </a>
    </Link>

    <div className="flex">
      <button className="mr-2 rounded px-3 text-slate-600 dark:text-white text-xs uppercase font-medium tracking-wider transition-colors hover:bg-slate-200 hover:dark:bg-slate-800">
        Login
      </button>
      <button className="bg-purple-600 rounded px-3 text-white text-xs uppercase font-medium tracking-wider transition-colors hover:bg-purple-500">
        Sign Up
      </button>
    </div>
  </nav>
);

export default NavBar;
