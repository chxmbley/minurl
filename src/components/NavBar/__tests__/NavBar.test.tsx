import { render } from '@testing-library/react';
import NavBar from '~components/NavBar';

test('renders a NavBar component', () => {
  const { container } = render(<NavBar />);
  expect(container).toMatchSnapshot();
});
