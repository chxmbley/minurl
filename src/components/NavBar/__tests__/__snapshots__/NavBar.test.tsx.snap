// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a NavBar component 1`] = `
<div>
  <nav
    class="flex justify-between py-4"
  >
    <a
      href="/"
    >
      <h1
        class="text-2xl font-bold text-slate-700 dark:text-slate-200"
      >
        minurl ðŸ’Ž
      </h1>
    </a>
    <div
      class="flex"
    >
      <button
        class="mr-2 button secondary"
      >
        Log In
      </button>
      <button
        class="button primary"
      >
        Sign Up
      </button>
    </div>
  </nav>
</div>
`;
